compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -446,23 +449,21 @@ fn run_compiler(
                     return early_exit();
                 }
 
-                tcx.analysis(())?;
+                let _ = tcx.analysis(());
```

Comment:
The `analysis` query returns a unit, so skipping returning of the query result doesn't save us anything. I think we do however need to make sure that the query gets executed (rather than just doing nothing if all dependencies are unchanged) to prevent query stealing from going wrong and I'm not sure if `.ensure()` does that.

Reply:
`analysis` is eval_always, so yes, it will be re-executed in any case.
`ensure()` is more about uniform style. When the return type is not (), `let _ = some query` should always be replaced by `ensure()` or `ensure_with_value()`, to make meaning explicit.